### PR TITLE
[RFC] vim-patch:7.4.1565

### PR DESCRIPTION
**Pull request by Shougo**
Problem:    Crash when assert_equal() runs into a NULL string.
Solution:   Check for NULL. (Dominique) Add a test.

https://github.com/vim/vim/commit/f1551964448607f8222de2d8f0992ea43eb2fe67

### CODE REVIEW

File: test/functional/legacy/assert_spec.lua [hunk 1]
```diff
@@ -62,6 +62,17 @@ describe('assert function:', function()
...
+
+    it('should change v:errors when expected is not equal to actual', function()
+      source([[
+      let s:v = {}
+      let s:x = {"a": s:v}
+      let s:v["b"] = s:x
+      let s:w = {"c": s:x, "d": ''}
+      call assert_equal(s:w, '')
+      ]])
+      expected_errors({"Expected NULL but got ''"})
```
Comment (jamessan) [inline test/functional/legacy/assert_spec.lua]: Neovim emits a different error here.

Comment (jamessan) [inline test/functional/legacy/assert_spec.lua]: Just change the string being matched in the expected_errors call.

**Comment by Shougo** [comment]
I have fixed the test.

**Comment by justinmk** [comment]
Builds are still failing.

**Comment by jamessan** [comment]
Merged.  Thanks!

**Comment by Shougo** [comment]
Thank you!

**Pull request merged**